lcidlc/src/Support.mm

```diff
@@ -535,15 +535,26 @@ static LCError LCValueArrayFromObjcArray(MCVariableRef var, NSArray *src)
 	t_error = kLCErrorNone;
 	
 	for(unsigned int t_index = 0; t_index < [src count] && t_error == kLCErrorNone; t_index++)
-	{
-		char t_key[12];
+    {
+        // SN-2015-02-17: [[ ExternalsApiV6 ]] Fix issues on index string creation:
+        //   - needs a dynamically allocated t_key,
+        //   - needs to pass the pointer to this key to variable_lookup_key
+        //   - sprintf format should be "%u", not "%ud"
+        char *t_key = nil;
+        t_key = new char[12];
```

Comment:
You don't need to do this... You do need to pass a pointer the variable. The correct thing to do is:

char t_key[12];
char *t_key_ptr;
t_key_ptr = t_key;

// Now you can pass &t_key_ptr.